### PR TITLE
audit/maker-1

### PR DESCRIPTION
**Pull request by exp7l**
### 1. Getting the `library` repo for the first time:
```
git clone git@github.com:0xMacro/library.git
git submodule init
git submodule update  # requires you have a github SSH key
npm install
npm run init
```

### 2. Checkout and run the Sommelier-3 PR branch:

```
git pull
git checkout audit/maker-1-final
npm run dev
```
## Todos

- reviewer: Please review the HTML rendered version, as formatting can be wrong.
- reviewer: For each of your own issues, please review if you like to change any wording/emphasis/formatting etc.
- reviewer: Drop the proposed changes in this PR, if any , and we can discuss in here.
- expo: When PR is ready for publishing
    - [ ] make TWAP lag notion report public
    - [ ] merge this PR into master and make sure this HTML version is public
    - [ ] pass it to the right person for tweeting

**References if helpful for reviewing**
- the version sent to Maker: https://www.notion.so/0xmacro/MakerDAO-1-Preliminary-Audit-Report-external-private-8a1480790d134bda8b86144b429cfdb8#ae5ce5844363445c8e51f5503d02796b
- the version and changelog for editing done by Jon the Editor: https://www.notion.so/0xmacro/MakerDAO-1-for-Copy-Editing-2d6efb17ba0240858608ce3a93b3d2a0#2dbf2113ad374b0187169e7a2325c958

## Report Preview URL

https://library-pr-21.onrender.com/library/audits/maker-1

### CODE REVIEW

File: client/library/library/audits/_layout.html [hunk 1]
```diff
@@ -398,6 +398,19 @@ <h2 class="IssueTitle mt-1 pl-3 text-xl sm:text-2xl font-bold">
                     {{Issues.statusNames[issue.status]}}
                     <svg xmlns="http://www.w3.org/2000/svg" class="px-1 h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
     <path stroke-linecap="round" stroke-linejoin="round" d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14" />
+  </svg>
+                  </a>
+                </else-if>
+                <else-if cond="issue.repoPath">
```
Comment (exp7l) [inline client/library/library/audits/_layout.html:404]: For adding a URL for "Addressed" issues.

File: lib/issues.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ export function validate(issue) {
   else if (issue.status === 'fixedprivate' && (issue.commit || issue.commitTree)) {
     throw new Error(`FixedPrivate issue cannot have 'commit' or 'commitTree' field (for issue: ${estimateIssuePreview(issue)})`)
   }
-  else if ((issue.commit || issue.commitTree) && issue.status !== 'fixed' && issue.status !== 'addressed') {
+  else if ((issue.commit || issue.commitTree || issue.repoPath) && issue.status !== 'fixed' && issue.status !== 'addressed') {
```
Comment (exp7l) [inline lib/issues.js:99]: For URL for "Addressed" issues.

File: content/collections/public/maker-1-issues.html [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        then spend enough on priority fees.
+        
+        Specifically, they will:
+        
+        1. take a flash loan,
+        2. swap WETH for MKR,
+        3. execute `fire()` on `KilnUniV3.sol`,
+        4. swap MKR back to WETH,
+        5. repay flash loan with fee `attackAmount` + `flashLoanFee`, and
+        6. profit (attacker) and loss (MakerDAO).
```
Comment (jonwalch) [inline content/collections/public/maker-1-issues.html:55]:
```suggestion
        1. Take a flash loan
        2. Swap WETH for MKR
        3. Execute `fire()` on `KilnUniV3.sol`
        4. Swap MKR back to WETH
        5. Repay flash loan with fee `attackAmount` + `flashLoanFee`
        6. Profit (attacker) and loss (MakerDAO).
```

File: client/library/library/audits/maker-1.html [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+              //----------End attack atomic execution-------------
+          
+              console.log("Attacker profit: %s", GemLike(WETH).balanceOf(address(attacker)));
+              console.log("Kiln receiver MKR Loss: %s ", (_est - GemLike(MKR).balanceOf(address(user))));
+              console.log("Kiln receiver MKR Loss: %s WAD", (_est - GemLike(MKR).balanceOf(address(user))) / WAD);
+          }
+          
+          ```
+          
+          ### [1] How the assumption `yen` is made
```
Comment (jonwalch) [inline client/library/library/audits/maker-1.html:164]: Maybe something like "Yen setting" or "Reason for yen setting"

File: client/library/library/audits/maker-1.html [hunk 5]
```diff
@@ -0,0 +1,454 @@
+<page
+  clientName="Maker"
+  reportDate="December 12th, 2022"
```
Comment (dlk61) [inline client/library/library/audits/maker-1.html:3]: Switch to the date that the report will be sent over to Maker

**Comment by exp7l** [comment]
https://library-pr-21.onrender.com/library/audits/maker-1

**Pull request merged**